providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/Util.java

```diff
@@ -1,36 +1,124 @@
-
 import dev.openfeature.sdk.exceptions.GeneralError;
+import io.grpc.ConnectivityState;
+import io.grpc.ManagedChannel;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 
 /**
- * Utils for flagd resolvers.
+ * Utility class for managing gRPC connection states and handling synchronization operations.
  */
+@Slf4j
 public class Util {
 
+    /**
+     * Private constructor to prevent instantiation of utility class.
+     */
     private Util() {
     }
 
     /**
-     * A helper to block the caller for given conditions.
-     * 
-     * @param deadline          number of milliseconds to block
-     * @param connectedSupplier func to check for status true
-     * @throws InterruptedException if interrupted
+     * A helper method to block the caller until a condition is met or a timeout occurs.
+     *
+     * @param deadline          the maximum number of milliseconds to block
+     * @param connectedSupplier a function that evaluates to {@code true} when the desired condition is met
+     * @throws InterruptedException if the thread is interrupted during the waiting process
+     * @throws GeneralError         if the deadline is exceeded before the condition is met
      */
-    public static void busyWaitAndCheck(final Long deadline, final Supplier<Boolean> connectedSupplier)
-            throws InterruptedException {
+    public static void busyWaitAndCheck(final Long deadline,
+                                        final Supplier<Boolean> connectedSupplier) throws InterruptedException {
         long start = System.currentTimeMillis();
 
         do {
             if (deadline <= System.currentTimeMillis() - start) {
-                throw new GeneralError(
-                        String.format("Deadline exceeded. Condition did not complete within the %d deadline",
-                                deadline));
+                throw new GeneralError(String.format("Deadline exceeded. Condition did not complete within the %d "
+                        + "deadline", deadline));
             }
 
             Thread.sleep(50L);
         } while (!connectedSupplier.get());
     }
+
+    /**
+     * Waits for a gRPC channel to reach a desired state within a specified timeout.
+     *
+     * @param ch           the gRPC {@link ManagedChannel} to monitor
+     * @param desiredState the desired {@link ConnectivityState} to wait for
+     * @param callback     a {@link Runnable} to execute when the desired state is reached
+     * @param timeout      the maximum time to wait
+     * @param unit         the {@link TimeUnit} for the timeout parameter
+     * @throws InterruptedException if the waiting thread is interrupted
+     * @throws GeneralError         if the deadline is exceeded before reaching the desired state
+     */
+    public static void waitForDesiredState(ManagedChannel ch, ConnectivityState desiredState, Runnable callback,
```

Comment:
```suggestion
    public static void waitForDesiredState(ManagedChannel channel, ConnectivityState desiredState, Runnable callback,
```

nit: var names like `ch` read more like Go than Java to me :sweat_smile: